src/github/dto/github.dto.ts

```diff
@@ -0,0 +1,73 @@
+import { ApiProperty } from '@nestjs/swagger';
+import {
+  IsIn,
+  IsNotEmpty,
+  IsNumber,
+  IsOptional,
+  IsString,
+} from 'class-validator';
+
+const events = [
+  'workflow_dispatch',
+  'repository_dispatch',
+  'check_run',
+  'check_suite',
+  'create',
+  'delete',
+  'deployment',
+  'deployment_status',
+  'fork',
+  'gollum',
+  'issue_comment',
+  'issues',
+  'label',
+  'milestone',
+  'page_build',
+  'project',
+  'project_card',
+  'project_column',
+  'public',
+  'pull_request',
+  'pull_request_review',
+  'pull_request_review_comment',
+  'pull_request_target',
+  'push',
+  'registry_package',
+  'release',
+  'status',
+  'watch',
+  'workflow_run',
+];
+
+export class GithubDTO {
+  @ApiProperty({ required: true })
+  @IsNotEmpty()
+  @IsString()
+  username: string;
+
+  @ApiProperty({ required: false })
+  @IsOptional()
+  @IsString()
+  bio: string;
+
+  @ApiProperty({ required: false })
+  @IsOptional()
+  @IsString()
+  avatarUrl: string;
+
+  @ApiProperty({ required: false })
+  @IsOptional()
+  @IsNumber()
+  followers: number;
+
+  @ApiProperty({ required: false })
+  @IsOptional()
+  @IsNumber()
+  repos: number;
+  // ... any other useful info public info https://docs.github.com/en/rest/reference/users
```

Comment:
This was a comment which was in the inital description of #44. Do we need some aditional info from Github or is the data we currently get enough?

Reply:
I had another look, I think these fields would be good too

- location
- company
- blog (I think this is their url they set)

Reply:
But this could come an another PR if you prefer

Reply:
Yes, these info would be good as well.
You want to display the location set in the Github Profile on a card? In other words do I need to geocode them as well?

Reply:
I'll add this for now, as these shouldnt be huge extensions to the current codebase

Reply:
Yep, the location will be geocoded to show on a map - but I think in this PR, only save the property `location.provided` and the geocoding data can come in another PR.

Maybe something like this...

```ts
{
   // ...
   location: {
      provided: string;
      lat: number;
      long: number;
   }
}
```

Reply:
Okay, then I'll just save the location that is sent for now.
I started with a geocoding service when I tried that out for #38 based on observables and rxjs. (I didnt open PR yet)
We could make a Commonsmodule or sth like that where we put such "helper-services"

Reply:
Yep, I don't think we need to allow them to save the `lat/lng`, this will come from the service after they provide a location string.

> I started with a geocoding service when I tried that out for #38 based on observables and rxjs. (I didnt open PR yet)

oh awesome! 👍 

> We could make a Commonsmodule or sth like that where we put such "helper-services"

I don't think this is needed for now, as this is the only place the location geocoding will be happening

Reply:
I've got one question over: 
In #38 theres also location for the discord data.
What would happen if a user has two different location set in his profiles?
That would probably result in wrong pins on the map. 
How should we handle this? Or do we even have to handle this?

Reply:
Sorry, that was my mistake 🤦‍♂️ 

We used to get it from Discord bot, but now we are planning to integrate with GitHub we don't need to get it from the Discord Bot anymore set by the user.

I create an issue for it to be removed from Discord module https://github.com/EddieHubCommunity/api/issues/66

Reply:
Okay, that will make it a bit easier for us then :D
You see #66 as new PR, or should I put it in this?

Reply:
I would prefer #66 as a new PR, the smaller the PRs the better

Reply:
Okay, then I implement the new properties
- organization
- blog
- location (not geocoded)

and start implementing geocoding service when this is merged
Sounds like a plan!

Reply:
Sounds good 👍